lib/Twig/NodeVisitor/Optimizer.php

```diff
@@ -97,6 +103,32 @@ private function optimizePrintNode(Twig_Node $node, Twig_Environment $env)
     }
 
     /**
+     * Transforms a "Concat" node with two "Constant" nodes into one "Constant" node.
+     *
+     * @return Twig_Node
+     */
+    private function optimizeConcatConst(Twig_Node $node, Twig_Environment $env)
+    {
+        if ($node instanceof Twig_Node_Expression_Binary_Concat) {
+            $leftNode = $node->getNode('left');
+            $rightNode = $node->getNode('right');
+            if (
+                $leftNode instanceof Twig_Node_Expression_Constant
+                && $rightNode instanceof Twig_Node_Expression_Constant
+            ) {
+                $leftNode->setAttribute(
```

Comment:
I would not mutate the left node in case it is shared in other nodes (which can happen in some cases). Create a new `Twig_Node_Expression_Constant` instead